python/composio/tools/base/runtime.py

```diff
@@ -376,7 +392,11 @@ def execute(request: BaseModel, metadata: t.Dict) -> BaseModel:
         if shell_argument is not None:
             kwargs[shell_argument] = metadata["workspace"].shells.recent
 
-        if auth_params > 0:
+        if connected_account_param:
+            kwargs["connected_account"] = (
+                _get_connected_account(app=app, entity_id=metadata["entity_id"]) or {}
+            )
+        if auth_param:
```

Comment:
The condition `if auth_params > 0:` was changed to `if auth_param:` which is more appropriate since it's a boolean flag. However, consider adding a type hint for this variable to make the boolean type explicit.